### ISSUE TITLE
Refactor QSO randomforest selection

### ISSUE THREAD

**Issue by dkirkby** [desihub/desitarget#482]
Most of the code in `isQSO_randomforest_north` and `isQSO_randomforest_south` is duplicated. This issue is a suggestion to refactor the common code into a shared function (or into the driver `isQSO_randomforest`).  The only differences between the north/south versions appear to be:
 - north calls `shift_photo_north` to translate to the DECaLS system.
 - north uses DR5-trained forest for release >= 6000, but south splits at 5000  (is this an intentional difference?)
 - the final cuts on the forest output probabilities are different:
```
<             pcut = np.where(tmp_r_Reduced > 20.,
<                             0.60 - (tmp_r_Reduced - 20.) * 0.08, 0.60)
<             pcut_HighZ = 0.42
---
>             pcut = np.where(tmp_r_Reduced > 20.8,
>                             0.83 - (tmp_r_Reduced - 20.8) * 0.025, 0.83)
>             pcut[tmp_r_Reduced > 21.5] = 0.8125 - 0.15 * (tmp_r_Reduced[tmp_r_Reduced > 21.5] - 21.5)
>             pcut[tmp_r_Reduced > 22.3] = 0.6925 - 0.70 * (tmp_r_Reduced[tmp_r_Reduced > 22.3] - 22.3)
>             pcut_HighZ = np.where(tmp_r_Reduced > 20.5,
>                                   0.55 - (tmp_r_Reduced - 20.5) * 0.025, 0.55)
```

**Comment by geordie666** [comment]
As much as possible, I try to let the people who are developing the target selection algorithms to write their code in their own style, even if that introduces redundancy. So, there is a balance between adhering strictly to Data Systems coding standards and utility for off-project contributors.

If @Cyeche wants me to refactor the code in this manner, he should respond to this issue. If he does not want me to refactor the code in this fashion, he should let me know and I'll close this issue.

**Comment by geordie666** [comment]
Addressed in #489.
